crypto/src/asset.rs

```diff
@@ -28,8 +34,62 @@ use crate::Fq;
 #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
 pub struct Id(pub Fq);
 
+/// Validator-specific unique assets, used for identifying staking tokens
+/// associated with a particular validator.
+pub struct ValidatorAsset {
+    validator_pubkey: PublicKey,
+}
+
+impl ValidatorAsset {
+    pub fn new(pk: PublicKey) -> Self {
+        ValidatorAsset {
+            validator_pubkey: pk,
+        }
+    }
+
+    pub fn denom(&self) -> Denom {
+        Denom(format!(
+            "d{}-{}",
+            NATIVE_ASSET_DENOM,
+            self.validator_pubkey
+                .to_bech32(PENUMBRA_BECH32_VALIDATOR_PREFIX)
+        ))
+    }
+
+    pub fn id(&self) -> Id {
+        Id::from(self.denom())
+    }
+}
+
+pub fn is_validator_denom(d: &Denom) -> bool {
+    let re = Regex::new(&format!(
+        r"^d{}-({}{}\w+)$",
+        NATIVE_ASSET_DENOM,
+        PENUMBRA_BECH32_VALIDATOR_PREFIX,
+        // Note, if tendermint-rs ever changes the bech32 separator from default,
+        // this will break
+        bech32::DEFAULT_SEPARATOR
+    ))
+    .unwrap();
+    let caps = re.captures(&d.0);
+    if caps.is_none() {
+        // There was no bech32 address captured
+        return false;
+    }
+
+    let bech32_address = caps.unwrap().get(1).map_or("", |m| m.as_str());
+
+    re.is_match(&d.0) && bech32::decode(bech32_address.to_lowercase()).is_ok()
+}
```

Comment:
This is doing the work of parsing the address but discarding it to return a bool, could it be refactored into an `impl FromStr` that would return the parsed data?  Then we could have `impl TryFrom<Denom>` that just calls `.from_str()` on the `Denom`'s inner string.

(I think it's better to put the parsing as a `FromStr` impl and have the `TryFrom<Denom>` forward to that rather than the other way around, since it means that we can use `.from_str()` to parse from any borrowed string, rather than just from an already-constructed `Denom`)